gnocchi/storage/_carbonara.py

```diff
@@ -117,9 +118,15 @@ def _store_metric_measures(metric, timestamp_key, aggregation,
                                granularity, data, offset=None, version=3):
         raise NotImplementedError
 
-    @staticmethod
-    def _list_split_keys_for_metric(metric, aggregation, granularity,
+    def _list_split_keys_for_metric(self, metric, aggregation, granularity,
                                     version=3):
+        return set(map(
+            functools.partial(carbonara.SplitKey, sampling=granularity),
+            utils.to_timestamps(self._list_split_keys(
```

Comment:
i'd prefer this added when needed. seems safer.

that way, we can (maybe) remove the temporary isinstance(value, numpy.datetime64) check in SplitKey and also the test changes? and if we need them, we can added them in the required commit (if actually needed).

disclaimer, my knowledge of this branch goes up to only https://github.com/gnocchixyz/gnocchi/pull/197/commits/706a9a5ce71c0dcf578cc5af728e3bf17bc389b0